packages/core/test/insured-bridge/e2e/helpers/MultiProviderHelper.js

```diff
@@ -0,0 +1,67 @@
+const hre = require("hardhat");
+
+const fs = require("fs");
+const path = require("path");
+
+function getProviders() {
+  return {
+    l1RpcProvider: new hre.ethers.providers.JsonRpcProvider("http://localhost:9545"),
+    l2RpcProvider: new hre.ethers.providers.JsonRpcProvider("http://localhost:8545"),
+  };
+}
+
+function getAllFiles(dirPath, arrayOfFiles = []) {
+  const files = fs.readdirSync(dirPath);
+
+  arrayOfFiles = arrayOfFiles || [];
+
+  files.forEach((file) => {
+    if (fs.statSync(dirPath + "/" + file).isDirectory()) arrayOfFiles = getAllFiles(dirPath + "/" + file, arrayOfFiles);
+    else arrayOfFiles.push(path.join(dirPath, "/", file));
+  });
+
+  return arrayOfFiles;
+}
+
+function findArtifactFromPaths(contractName, artifactsPath) {
```

Comment:
Are we doing all of this because these strategies won't work for optimism artifacts?

https://hardhat.org/guides/compile-contracts.html#reading-artifacts

The multiple sets of artifacts breaks a lot of assumptions. How often do you think we'll need to work around this. For instance, we can't really export multiple sets of artifacts with our current infrastructure -- is exporting optimism-specific artifacts something we should worry about?

Reply:
correct. the whole point of this is to let the importer explicitly choose which compilation environment to import the contacts from (`artifacts` or `artifacts-ovm` depending on the file.

In general, this might become something we need to do and factor into how we export artifacts. I can imagine a world where the `contracts-node` & `contracts-frontend` methods have a second prop that lets you define the EVM compatible compilation enviroment. i.e `optimism` as a selectable parameter, which will return the artifact compiled for that environment. For now, this is unlikely to be worth it but in the future, it might be required, especially if we have a whole set of contracts deployed on these networks.

Tracking even future-forward, it is possible (and somewhat likely) that there will be other L2's that have slightly modified opcodes & compilation processes that would necessitate this kind of exporting. 

For now though, I think it's fine to isolate this behavior to testing and we can figure out a more robust solution later. Especially seeing the export techniques are still somewhat in flux (with your PR #3229 for example).